src/EventStore.Core/Certificates/OptionsCertificateProvider.cs

```diff
@@ -19,6 +19,14 @@ public class OptionsCertificateProvider: CertificateProvider {
 
 			var (certificate, intermediates) = _options.LoadNodeCertificate();
 
+			var certificateCN = certificate.GetCommonName();
+			var reservedNodeCN = _options.Certificate.CertificateReservedNodeCommonName;
+
+			if (certificateCN != reservedNodeCN)
+				Log.Warning(
```

Comment:
The server should shut down when these don't match rather than continuing to run with an invalid configuration